java/java.hints/src/org/netbeans/modules/java/hints/errors/Utilities.java

```diff
@@ -3114,16 +3114,30 @@ public Void visitClass(ClassTree node, Void p) {
     public static boolean isCompatibleWithSwitchExpression(SwitchTree st) {
         boolean firstCase = true;
         Name leftTreeName = null;
-        for (CaseTree ct : st.getCases()) {
+        int CaseCount =0;
+        List<? extends CaseTree> cases = st.getCases(); 
+        
+        for (CaseTree ct : cases) {
+           CaseCount++;
             List<StatementTree> statements = new ArrayList<>(ct.getStatements());
             switch (statements.size()) {
                 case 0:
                     break;
                 case 1:
                     if (firstCase && leftTreeName == null && statements.get(0).getKind() == Tree.Kind.RETURN) {
                         break;
-                    } else {
-                        return false;
+                    } else if (CaseCount == cases.size() && statements.get(0).getKind() == Tree.Kind.EXPRESSION_STATEMENT) {
```

Comment:
Majority of the code in case 1 is same as case 2, can reuse?

Reply:
had  reused the common code in last commit